src/OpenTelemetry.Instrumentation.Http/HttpClientInstrumentation.cs

```diff
@@ -43,14 +43,7 @@ public HttpClientInstrumentation(HttpClientTraceInstrumentationOptions options)
         // the framework will fall back to creating activity anyways due to active diagnostic source listener
         // To prevent this, isEnabled is implemented which will return false always
         // so that the sampler's decision is respected.
-        if (HttpHandlerDiagnosticListener.IsNet7OrGreater)
-        {
-            this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabledNet7OrGreater, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
-        }
-        else
-        {
-            this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabled, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
-        }
+        this.diagnosticSourceSubscriber = HttpHandlerDiagnosticListener.IsNet7OrGreater ? new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabledNet7OrGreater, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent) : new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabled, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
```

Comment:
nit: IMO these conditional statements are best as multi-liners. But this one is super long and hard to read 🤣

```suggestion
        this.diagnosticSourceSubscriber = HttpHandlerDiagnosticListener.IsNet7OrGreater
                ? new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabledNet7OrGreater, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent)
                : new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabled, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
```